### PR TITLE
drivers/mrf24j40 add tx retries needed get operation

### PR DESCRIPTION
**Pull request by bergzand**
requires #7447

### CODE REVIEW

**Comment by bergzand** [comment]
Rebased because #7447 was merged. No longer depends on another PR.

**Comment by bergzand** [comment]
Rebased to resolve merge conflicts

File: drivers/mrf24j40/include/mrf24j40_registers.h [hunk 1]
```diff
@@ -268,6 +268,14 @@ extern "C" {
 /** @} */
 
 /**
+ * @name    Shift offsets for TXSTAT register (0x24)
+ * @{
+ */
+#define MRF24J40_TXSTAT_MAX_FRAME_RETRIES_SHIFT 6
```
Comment (smlng) [inline drivers/mrf24j40/include/mrf24j40_registers.h]: please put in parentheses and make explicitly unsigned, i.e `(6U)` - might not be necessary but would be more consistent with (majority) of RIOT code base.

File: drivers/mrf24j40/include/mrf24j40_registers.h [hunk 2]
```diff
@@ -268,6 +268,14 @@ extern "C" {
...
+#define MRF24J40_TXSTAT_CCAFAIL_SHIFT           5
```
Comment (smlng) [inline drivers/mrf24j40/include/mrf24j40_registers.h]: not used (yet?)

File: drivers/mrf24j40/mrf24j40_netdev.c [hunk 3]
```diff
@@ -538,6 +547,7 @@ static void _isr(netdev_t *netdev)
 #ifdef MODULE_NETSTATS_L2
         if (netdev->event_callback && (dev->netdev.flags & MRF24J40_OPT_TELL_TX_END)) {
             uint8_t txstat = mrf24j40_reg_read_short(dev, MRF24J40_REG_TXSTAT);
+	    dev->tx_retries = (txstat >> MRF24J40_TXSTAT_MAX_FRAME_RETRIES_SHIFT);
```
Comment (smlng) [inline drivers/mrf24j40/mrf24j40_netdev.c]: looks like a `tab` here? Change to spaces.

File: drivers/mrf24j40/include/mrf24j40_registers.h (hunk 2 above)
Comment (bergzand) [reply drivers/mrf24j40/include/mrf24j40_registers.h]: Correct, I added it for completeness of that register definition

Comment (smlng) [reply drivers/mrf24j40/include/mrf24j40_registers.h]: fine by me, just asking.

File: drivers/mrf24j40/include/mrf24j40_registers.h (hunk 1 above)
Comment (bergzand) [reply drivers/mrf24j40/include/mrf24j40_registers.h]: Ack, fixed

**Comment by bergzand** [comment]
@smlng Thank you for you comments, I think I've addressed them all.

**Comment by smlng** [comment]
ACK, please squash

**Comment by bergzand** [comment]
squashed

**Comment by smlng** [comment]
Time to move this forward: ACK and GO!

**Pull request merged**